paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicStyleResourceItem.kt

```diff
@@ -0,0 +1,89 @@
+ * limitations under the License.
+ */
+package app.cash.paparazzi.internal.resources.base
+
+import app.cash.paparazzi.internal.resources.ResourceSourceFile
+import com.android.ide.common.rendering.api.ResourceNamespace
+import com.android.ide.common.rendering.api.ResourceReference
+import com.android.ide.common.rendering.api.StyleItemResourceValue
+import com.android.ide.common.rendering.api.StyleResourceValue
+import com.android.resources.ResourceType
+import com.android.resources.ResourceType.ATTR
+import com.android.resources.ResourceVisibility
+import com.google.common.collect.ImmutableTable
+import com.google.common.collect.Table
+import java.util.Collections
+import java.util.logging.Logger
+
+/**
+ * Ported from: [BasicStyleResourceItem.java](https://cs.android.com/android-studio/platform/tools/base/+/f3c4ef558ce2a3ac9eacf7007aee8f6e056235eb:resource-repository/main/java/com/android/resources/base/BasicStyleResourceItem.java)
+ *
+ * Resource item representing a style resource.
+ */
+class BasicStyleResourceItem(
+  name: String,
+  sourceFile: ResourceSourceFile,
+  visibility: ResourceVisibility,
+  private val parentStyleName: String?,
+  styleItems: Collection<StyleItemResourceValue>
+) : BasicValueResourceItemBase(ResourceType.STYLE, name, sourceFile, visibility),
+  StyleResourceValue {
+  /** Style items keyed by the namespace and the name of the attribute they define.  */
+  private val styleItemTable: Table<ResourceNamespace, String, StyleItemResourceValue>
+
+  init {
+    val tableBuilder: ImmutableTable.Builder<ResourceNamespace, String, StyleItemResourceValue> =
+      ImmutableTable.builder()
+    val duplicateCheckMap = mutableMapOf<ResourceReference, StyleItemResourceValue>()
+    for (styleItem in styleItems) {
+      val attr: ResourceReference? = styleItem.attr
+      if (attr != null) {
+        // Check for duplicate style item definitions. Such duplicate definitions are present in the framework resources.
+        val previouslyDefined: StyleItemResourceValue? = duplicateCheckMap.put(attr, styleItem)
+        if (previouslyDefined == null) {
+          tableBuilder.put(attr.namespace, attr.name, styleItem)
+        } else if (previouslyDefined != styleItem) {
+          LOG.warning("Conflicting definitions of \"${styleItem.attrName}\" in style \"$name\"")
+        }
+      }
+    }
+    styleItemTable = tableBuilder.build()
+  }
+
+  override fun getParentStyleName(): String? = parentStyleName
+
+  override fun getItem(namespace: ResourceNamespace, name: String): StyleItemResourceValue? = styleItemTable.get(namespace, name)
+
+  override fun getItem(attr: ResourceReference): StyleItemResourceValue? {
+    assert(attr.resourceType === ATTR)
+    return styleItemTable.get(attr.namespace, attr.name)
+  }
+
+  override fun getDefinedItems(): MutableCollection<StyleItemResourceValue> {
+    return Collections.unmodifiableCollection(styleItemTable.values())
+  }
```

Comment:
```suggestion
  override fun getDefinedItems(): Collection<StyleItemResourceValue> = styleItemTable.values()
```

Table.values() returns an `ImmutableCollection`.